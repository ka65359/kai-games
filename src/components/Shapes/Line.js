////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////
/**
 * @author Kai
 * @version 1.0.0
 * @exports Line
 *
 * @module Line
 * @description A module that builds a line represented on a game board that can be styled
 *  and have a callback function when clicked.
 * @typedef {Object} Line
 * @property {function} onClick - The function that fires when a line is clicked
 * @property {'top'|'left'|'right'|'bottom'} position - The position of the border line of a
 *  game board grid square
 * @property {string} lineColor - The color of the line. Takes a CSS.color string or hex value
 */
////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////
import React from "react";
import PropTypes from "prop-types";
import { color } from "@storybook/addon-knobs";
import "./line.scss";

const Line = ({ onClick, position, lineColor }) => {
  if (typeof onClick !== "function") {
    onClick = () => {};
  }

  let className = "kai-line line";
  if (position) {
    className += " kai-shape-pos-" + position;
  }

  let style = {};
  if (color("color", lineColor)) {
    style.color = lineColor;
  }

  return <div className={className} onClick={onClick} style={style}></div>;
};

Line.propTypes = {
  onClick: PropTypes.func,
  position: PropTypes.string,
  lineColor: PropTypes.string
};

export default Line;
