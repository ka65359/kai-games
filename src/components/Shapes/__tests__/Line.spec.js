import React from "react";
import { shallow } from "enzyme";
import Line from "../Line";

describe("Line() tests", () => {
  let data;

  beforeEach(function() {
    data = {
      position: "top",
      lineColor: "#000",
      onClick: jest.fn()
    };
  });

  afterEach(function() {
    data = null;
  });

  it("should render correctly in mode", () => {
    const component = shallow(<Line {...data} />);

    expect(component).toMatchSnapshot();
  });
});
