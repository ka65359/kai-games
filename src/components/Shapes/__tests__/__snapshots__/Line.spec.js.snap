// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line() tests should render correctly in mode 1`] = `ShallowWrapper {}`;
